src/main/java/net/mcreator/ui/dialogs/NewModElementDialog.java

```diff
@@ -56,12 +66,15 @@ public static void showNameDialog(MCreator mcreator, ModElementType<?> type) {
 								L10N.t("common.mod_element_name")).validate();
 					}
 				}, L10N.t("dialog.new_modelement.create_new", type.getReadableName()),
-				UIManager.getString("OptionPane.cancelButtonText"), null, regName);
+				UIManager.getString("OptionPane.cancelButtonText"), null,
+				PanelUtils.northAndCenterElement(new JScrollPane(breadcrumb, JScrollPane.VERTICAL_SCROLLBAR_NEVER,
+						JScrollPane.HORIZONTAL_SCROLLBAR_ALWAYS), regName));
 
 		if (modName != null && !modName.isEmpty()) {
 			modName = JavaConventions.convertToValidClassName(modName);
 
 			ModElement element = new ModElement(mcreator.getWorkspace(), modName, type);
+			element.setParentFolder(breadcrumb.getFolder());

```

Comment:
You should likely avoid calling this if root - no setting = assumed root

Reply:
Just added this locally.